python/cudaq/kernel/ast_bridge.py

```diff
@@ -960,6 +960,26 @@ def bodyBuilder(iterVar):
                 self.__applyQuantumOperation(node.func.id, [], qubitTargets)
                 return
 
+            if node.func.id in ["ch", "cx", "cy", "cz", "cs", "ct"]:
+                # These are single target controlled quantum operation
+                target = self.popValue()
+                # Should be number of arguments minus one for the controls
+                controls = [self.popValue() for i in range(len(node.args) - 1)]
```

Comment:
Here we should check that self.valueStack has 2 elements and be careful about slurping up all values when these operations only support a single control and a single target.

Reply:
Fixed in [dd363e9](https://github.com/NVIDIA/cuda-quantum/pull/1279/commits/dd363e9bd59b3622c5dc053ef61ea3374f6d7a18)